src/pip/_internal/cli/cmdoptions.py

```diff
@@ -840,6 +840,8 @@ def _handle_no_use_pep517(option, opt, value, parser):
     action='store_true',
     help=SUPPRESS_HELP,
 )  # type: Callable[..., Option]
+# This way we know we can ignore this option in various docs auto generation
+setattr(always_unzip, 'deprecated', True)
```

Comment:
Is there a particlar reason to use `setattr` here? If it’s to avoid linting errors, I would instead do

```python
class DeprecatedOptionFactory(partial):
    deprecated = True

always_unzip = DeprecatedOptionFactory(
    Option,
    ...
)
```

Reply:
Urrgh, because `functools.partial` is a generic class in Mypy… this sucks. I don’t really know how to fix this in a Python 2-compatible way. @pradyunsg any ideas?

Reply:
Bah, I'd keep this as is, file an issue about this and link it as To-Do-with #4659.

Reply:
I will create an issue after we merge this, so I can permalink to the code

Reply:
I will create an issue after we merge this, so I can permalink to the code

Reply:
Hi @NoahGorny 

In commit https://github.com/pypa/pip/commit/17b1c76ff547d59d6d578bad6c1b9e9cd0ae65ee , we got rid of the code in question, so you won't need to create an isue. You can go ahead and resolve this comment too.